openshift_tools/reporting/zabbixplot.py

```diff
@@ -52,41 +69,47 @@ def get_event_data(self):
-                                         triggerids=triggerids, limit=self.limit, selectHosts="extend",
-                                         sortfield="clock", sortorder="DESC")
-
+        self.events = self.zapi.event.get(time_from=self.timespan.timestamp_start,
+                                          time_till=self.timespan.timestamp_end, triggerids=triggerids,
+                                          limit=self.limit, selectHosts="extend", sortfield="clock", sortorder="DESC")
+        if len(self.events) >= self.limit:
+            print("Warning! Data has been truncated by the result limit. The graph will be inaccurate.",
+                  file=sys.stderr)
+
+    # Reason: All of these parameters are needed
+    # Status: permanently disabled
+    # pylint: disable=too-many-arguments
     def __init__(self, server, username, password, group, trigger, days, limit):
-        self.server = server
-        self.username = username
-        self.password = password
         self.group = group
         self.trigger = trigger
         self.days = days
         self.limit = limit
 
-        self.time_start = datetime.datetime.now() - datetime.timedelta(days=days)
-        self.time_end = datetime.datetime.now() - datetime.timedelta(days=0)
-        self.timestamp_start = int(self.time_start.strftime("%s"))
-        self.timestamp_end = int(self.time_end.strftime("%s"))
-        self.mtime_start = mdates.date2num(self.time_start)
-        self.mtime_end = mdates.date2num(self.time_end)
+        self.timespan = ZabbixTimespan(days)
 
-        self.zapi = ZabbixAPI(self.server)
-        self.zapi.login(self.username, self.password)
+        self.zapi = ZabbixAPI(server)
+        self.zapi.login(username, password)
         self.events = False
 
-    def plot_aggregate(self, periods, num_hosts):
-        """Plots a bar chart from aggregated event occurences"""
-        self.get_event_data()
-
-        period_count = (periods or self.days) + 1  # One more for the ending pseudo-period
-        periods = np.linspace(self.timestamp_start, self.timestamp_end, num=period_count, endpoint=True)
-        period_seconds = (self.timestamp_end - self.timestamp_start) / (period_count - 1)
-
-        host_func = lambda event: event["hosts"][0]["hostid"]
+    @staticmethod
+    def _calculate_event_downtime(event_value, event_clock, current_period_end, down):
+        """Calculate the amount of downtime or uptime caused by an event"""
+        deltatime = 0
+        # TODO: This will ignore events that start before the data
+        if event_value != "0":
+            # Add downtime, assuming this may be the last event
+            deltatime = current_period_end - event_clock
+            down = True
+        elif event_value == "0" and down:
+            # Remove downtime, compensating for recovered downtime
+            deltatime = current_period_end - event_clock
+            down = False
+        return down, deltatime
```

Comment:
"down" is not in scope. You should initialize "down" to a value at the beginning of the method so that it's in scope here.